In [1]:
import sys

# 10818
- 문제

N개의 정수가 주어진다. 이때, 최솟값과 최댓값을 구하는 프로그램을 작성하시오.
- 입력

첫째 줄에 정수의 개수 N (1 ≤ N ≤ 1,000,000)이 주어진다. 둘째 줄에는 N개의 정수를 공백으로 구분해서 주어진다. 모든 정수는 -1,000,000보다 크거나 같고, 1,000,000보다 작거나 같은 정수이다.
- 출력

첫째 줄에 주어진 정수 N개의 최솟값과 최댓값을 공백으로 구분해 출력한다.
- 예제 입력 1

5

20 10 35 30 7

- 예제 출력 1 

7 35

In [1]:
a=int(input())
b=list(map(int, input().split()))
print(min(b), max(b))

 123
 456


456 456


# 2562
- 문제

9개의 서로 다른 자연수가 주어질 때, 이들 중 최댓값을 찾고 그 최댓값이 몇 번째 수인지를 구하는 프로그램을 작성하시오.

예를 들어, 서로 다른 9개의 자연수

3, 29, 38, 12, 57, 74, 40, 85, 61

이 주어지면, 이들 중 최댓값은 85이고, 이 값은 8번째 수이다.

- 입력

첫 째 줄부터 아홉 번째 줄까지 한 줄에 하나의 자연수가 주어진다. 주어지는 자연수는 100 보다 작다.

- 출력

첫째 줄에 최댓값을 출력하고, 둘째 줄에 최댓값이 몇 번째 수인지를 출력한다.

- 예제 입력 1 

3
29
38
12
57
74
40
85
61


- 예제 출력 1 

85
8

In [110]:
list1=[]
for i in range(9):
    a=int(input())
    list1.append(a)
print(max(list1))
print(list1.index(max(list1))+1)

3
29
38
12
57
74
40
85
61
85
8


## 2577
- 문제

세 개의 자연수 A, B, C가 주어질 때 A×B×C를 계산한 결과에 0부터 9까지 각각의 숫자가 몇 번씩 쓰였는지를 구하는 프로그램을 작성하시오.

예를 들어 A = 150, B = 266, C = 427 이라면 

A × B × C = 150 × 266 × 427 = 17037300 이 되고, 

계산한 결과 17037300 에는 0이 3번, 1이 1번, 3이 2번, 7이 2번 쓰였다.

- 입력

첫째 줄에 A, 둘째 줄에 B, 셋째 줄에 C가 주어진다. A, B, C는 모두 100보다 같거나 크고, 1,000보다 작은 자연수이다.

- 출력

첫째 줄에는 A×B×C의 결과에 0 이 몇 번 쓰였는지 출력한다. 마찬가지로 둘째 줄부터 열 번째 줄까지 A×B×C의 결과에 1부터 9까지의 숫자가 각각 몇 번 쓰였는지 차례로 한 줄에 하나씩 출력한다.

- 예제 입력 1

150
266
427
- 예제 출력 1 

3
1
0
2
0
0
0
2
0
0

In [62]:
a1=int(input())
a2=int(input())
a3=int(input())
a=str(a1*a2*a3)

a=[int(i) for i in a]

for i in range(10):
    print(a.count(i))

150
266
427
3
1
0
2
0
0
0
2
0
0


## 3052
- 문제

두 자연수 A와 B가 있을 때, A%B는 A를 B로 나눈 나머지 이다. 예를 들어, 7, 14, 27, 38을 3으로 나눈 나머지는 1, 2, 0, 2이다. 

수 10개를 입력받은 뒤, 이를 42로 나눈 나머지를 구한다. 그 다음 서로 다른 값이 몇 개 있는지 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄부터 열번째 줄 까지 숫자가 한 줄에 하나씩 주어진다. 이 숫자는 1,000보다 작거나 같고, 음이 아닌 정수이다.

- 출력

첫째 줄에, 42로 나누었을 때, 서로 다른 나머지가 몇 개 있는지 출력한다.

- 예제 입력 1 

39
40
41
42
43
44
82
83
84
85
- 예제 출력 1 

6
- 힌트

39, 40, 41, 42, 43, 44, 82, 83, 84, 85를 42로 나눈 나머지는 39, 40, 41, 0, 1, 2, 40, 41, 0, 1이다. 서로 다른 값은 모두 6개가 있다.

In [77]:
print(len(set([int(input()) % 42 for i in range(10)])))

39
40
41
42
43
44
82
83
84
85
6


## 1546
- 문제

세준이는 기말고사를 망쳤다. 세준이는 점수를 조작해서 집에 가져가기로 했다. 일단 세준이는 자기 점수 중에 최댓값을 골랐다. 이 값을 M이라고 한다. 그리고 나서 모든 점수를 점수/M*100으로 고쳤다.

예를 들어, 세준이의 최고점이 70이고, 수학점수가 50이었으면 수학점수는 50/70*100이 되어 71.43점이 된다.

세준이의 성적을 위의 방법대로 새로 계산했을 때, 새로운 평균을 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 시험 본 과목의 개수 N이 주어진다. 이 값은 1000보다 작거나 같다. 둘째 줄에 세준이의 현재 성적이 주어진다. 이 값은 100보다 작거나 같은 음이 아닌 정수이고, 적어도 하나의 값은 0보다 크다.

- 출력

첫째 줄에 새로운 평균을 출력한다. 정답과의 절대/상대 오차는 10-2까지 허용한다.

- 예제 입력 1 

3
40 80 60
- 예제 출력 1

75.00

- 예제 입력 2 

3

10 20 30
- 예제 출력 2 

66.666667

In [3]:
n=int(input())

a=list(map(int, input().split()))

a=[i/max(a)*100 for i in a]

print(sum(a)/n)

3
40 60 80


75.0

## 8958
- 문제

"OOXXOXXOOO"와 같은 OX퀴즈의 결과가 있다. O는 문제를 맞은 것이고, X는 문제를 틀린 것이다. 문제를 맞은 경우 그 문제의 점수는 그 문제까지 연속된 O의 개수가 된다. 예를 들어, 10번 문제의 점수는 3이 된다.

"OOXXOXXOOO"의 점수는 1+2+0+0+1+0+0+1+2+3 = 10점이다.

OX퀴즈의 결과가 주어졌을 때, 점수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 테스트 케이스의 개수가 주어진다. 각 테스트 케이스는 한 줄로 이루어져 있고, 길이가 0보다 크고 80보다 작은 문자열이 주어진다. 문자열은 O와 X만으로 이루어져 있다.

- 출력

각 테스트 케이스마다 점수를 출력한다.

- 예제 입력 1 

5

OOXXOXXOOO

OOXXOOXXOO

OXOXOXOXOXOXOX

OOOOOOOOOO

OOOOXOOOOXOOOOX
- 예제 출력 1 

10
9
7
55
30

In [ ]:
b=int(input())
OX = [input() for _ in range(b)]

b=[]
for ox in OX:
    count = 0
    b.clear()
    for i in range(len(ox)):
        if ox[i]=='O':
            count += 1
            b.append(count)
        elif ox[i]=='X':
            count = 0
    print(sum(b))

## 4344
- 문제

대학생 새내기들의 90%는 자신이 반에서 평균은 넘는다고 생각한다. 당신은 그들에게 슬픈 진실을 알려줘야 한다.

- 입력

첫째 줄에는 테스트 케이스의 개수 C가 주어진다.

둘째 줄부터 각 테스트 케이스마다 학생의 수 N(1 ≤ N ≤ 1000, N은 정수)이 첫 수로 주어지고, 이어서 N명의 점수가 주어진다. 점수는 0보다 크거나 같고, 100보다 작거나 같은 정수이다.

- 출력

각 케이스마다 한 줄씩 평균을 넘는 학생들의 비율을 반올림하여 소수점 셋째 자리까지 출력한다.

- 예제 입력 1 
```
5
5 50 50 70 80 100
7 100 95 90 80 70 60 50
3 70 90 80
3 70 90 81
9 100 99 98 97 96 95 94 93 91
```
- 예제 출력 1 
```
40.000%
57.143%
33.333%
66.667%
55.556%
```

In [24]:
num=int(input())
A=[list(map(int,input().split())) for _ in range(num)]

for a in A:
    a.pop(0)
    count=0
    for v in a :
        if v>(sum(a)/len(a)):
            count += 1
    per=count/len(a)*100
    print('%.3f'%per+'%')

5
5 50 50 70 80 100
7 100 95 90 80 70 60 50
3 70 90 80
3 70 90 81
9 100 99 98 97 96 95 94 93 91


## 15596
정수 n개가 주어졌을 때, n개의 합을 구하는 함수를 작성하시오.

a: 합을 구해야 하는 정수 n개가 저장되어 있는 리스트 
    (0 ≤ a[i] ≤ 1,000,000, 1 ≤ n ≤ 3,000,000)

리턴값: a에 포함되어 있는 정수 n개의 합 (정수)

In [36]:
def suma(a):
    s=0
    for v in a:
        s += v
    return s

a=[1,2,3,4,5,6]
suma(a)

21

## 4673
- 문제

셀프 넘버는 1949년 인도 수학자 D.R. Kaprekar가 이름 붙였다. 양의 정수 n에 대해서 d(n)을 n과 n의 각 자리수를 더하는 함수라고 정의하자. 예를 들어, d(75) = 75+7+5 = 87이다.

양의 정수 n이 주어졌을 때, 이 수를 시작해서 n, d(n), d(d(n)), d(d(d(n))), ...과 같은 무한 수열을 만들 수 있다. 

예를 들어, 33으로 시작한다면 다음 수는 33 + 3 + 3 = 39이고, 그 다음 수는 39 + 3 + 9 = 51, 다음 수는 51 + 5 + 1 = 57이다. 이런식으로 다음과 같은 수열을 만들 수 있다.

33, 39, 51, 57, 69, 84, 96, 111, 114, 120, 123, 129, 141, ...

n을 d(n)의 생성자라고 한다. 위의 수열에서 33은 39의 생성자이고, 39는 51의 생성자, 51은 57의 생성자이다. 생성자가 한 개보다 많은 경우도 있다. 예를 들어, 101은 생성자가 2개(91과 100) 있다. 

생성자가 없는 숫자를 셀프 넘버라고 한다. 100보다 작은 셀프 넘버는 총 13개가 있다. 1, 3, 5, 7, 9, 20, 31, 42, 53, 64, 75, 86, 97

10000보다 작거나 같은 셀프 넘버를 한 줄에 하나씩 출력하는 프로그램을 작성하시오.

입력은 없다.

- 출력
10,000보다 작거나 같은 셀프 넘버를 한 줄에 하나씩 증가하는 순서로 출력한다.

- 예제 출력 1 
```
1
3
5
7
9
20
31
42
53
64
 |
 |       <-- a lot more numbers
 |
9903
9914
9925
9927
9938
9949
9960
9971
9982
9993
```

In [117]:
def selfnum(n):
    a=[]
    a.append(n)
    a.extend((int(x) for x in str(n)))
    return sum(a)

b=[selfnum(i) for i in range(9973)]

b=set(b)
c=[i for i in range(10001)]
c=set(c)

c=list(c-b)

c.sort()

for v in c:
    print(v)

1
3
5
7
9
20
31
42
53
64
75
86
97
108
110
121
132
143
154
165
176
187
198
209
211
222
233
244
255
266
277
288
299
310
312
323
334
345
356
367
378
389
400
411
413
424
435
446
457
468
479
490
501
512
514
525
536
547
558
569
580
591
602
613
615
626
637
648
659
670
681
692
703
714
716
727
738
749
760
771
782
793
804
815
817
828
839
850
861
872
883
894
905
916
918
929
940
951
962
973
984
995
1006
1021
1032
1043
1054
1065
1076
1087
1098
1109
1111
1122
1133
1144
1155
1166
1177
1188
1199
1210
1212
1223
1234
1245
1256
1267
1278
1289
1300
1311
1313
1324
1335
1346
1357
1368
1379
1390
1401
1412
1414
1425
1436
1447
1458
1469
1480
1491
1502
1513
1515
1526
1537
1548
1559
1570
1581
1592
1603
1614
1616
1627
1638
1649
1660
1671
1682
1693
1704
1715
1717
1728
1739
1750
1761
1772
1783
1794
1805
1816
1818
1829
1840
1851
1862
1873
1884
1895
1906
1917
1919
1930
1941
1952
1963
1974
1985
1996
2007
2022
2033
2044
2055
2066
2077
2088
2099
2110
2112
2123
2134
2145
2156
2167
2178
2189
2200
2211
2213
2224
2235
2246


### 1065
- 문제

어떤 양의 정수 X의 자리수가 등차수열을 이룬다면, 그 수를 한수라고 한다. 등차수열은 연속된 두 개의 수의 차이가 일정한 수열을 말한다. N이 주어졌을 때, 1보다 크거나 같고, N보다 작거나 같은 한수의 개수를 출력하는 프로그램을 작성하시오. 

- 입력

첫째 줄에 1,000보다 작거나 같은 자연수 N이 주어진다.

- 출력

첫째 줄에 1보다 크거나 같고, N보다 작거나 같은 한수의 개수를 출력한다.
```
예제 입력 1 
110
예제 출력 1 
99
예제 입력 2 
1
예제 출력 2 
1
```

In [146]:
num = int(input())
cnt=0
for n in range(1,num+1):
    n=str(n)
    if len(n)==1 or len(n)==2:
        cnt += 1
    else :
        if int(n[0])-int(n[1])==int(n[1])-int(n[2]):
            cnt += 1    
print(cnt)

110
99


## 11654
- 문제
알파벳 소문자, 대문자, 숫자 0-9중 하나가 주어졌을 때, 주어진 글자의 아스키 코드값을 출력하는 프로그램을 작성하시오.

- 입력
알파벳 소문자, 대문자, 숫자 0-9 중 하나가 첫째 줄에 주어진다.

- 출력
입력으로 주어진 글자의 아스키 코드 값을 출력한다.
```
예제 입력 1 
A
예제 출력 1 
65
예제 입력 2 
C
예제 출력 2 
67
```

In [17]:
a=input()

if type(a)==str:
    print(ord(a))
else :
    print(chr(a))

7
55


## 11720

- 문제

N개의 숫자가 공백 없이 쓰여있다. 이 숫자를 모두 합해서 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 숫자의 개수 N (1 ≤ N ≤ 100)이 주어진다. 둘째 줄에 숫자 N개가 공백없이 주어진다.

- 출력

입력으로 주어진 숫자 N개의 합을 출력한다.
```
예제 입력 1 
1
1
예제 출력 1 
1
예제 입력 2 
5
54321
예제 출력 2 
15
```

In [27]:
a=input()
b=str(input())
c=0
for i in b:
    c+=int(i)
print(c)

5
12345
15


## 10809
- 문제

알파벳 소문자로만 이루어진 단어 S가 주어진다. 각각의 알파벳에 대해서, 단어에 포함되어 있는 경우에는 처음 등장하는 위치를, 포함되어 있지 않은 경우에는 -1을 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 단어 S가 주어진다. 단어의 길이는 100을 넘지 않으며, 알파벳 소문자로만 이루어져 있다.

- 출력

각각의 알파벳에 대해서, a가 처음 등장하는 위치, b가 처음 등장하는 위치, ... z가 처음 등장하는 위치를 공백으로 구분해서 출력한다.

만약, 어떤 알파벳이 단어에 포함되어 있지 않다면 -1을 출력한다. 단어의 첫 번째 글자는 0번째 위치이고, 두 번째 글자는 1번째 위치이다.
```
예제 입력 1 
baekjoon
예제 출력 1 
1 0 -1 -1 2 -1 -1 -1 -1 4 3 -1 -1 7 5 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
```

In [ ]:
import string

In [79]:
abc=list(string.ascii_lowercase)
a=input()

for i,v in enumerate(a):
    if v in abc:
        abc[abc.index(v)]=i
        
for x in abc:
    if type(x)==str:
        abc[abc.index(x)]=-1
#     print(x)

for x in abc:
    print(x)

backjoon
1
0
2
-1
-1
-1
-1
-1
-1
4
3
-1
-1
7
5
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1
-1


## 2675
- 문제

문자열 S를 입력받은 후에, 각 문자를 R번 반복해 새 문자열 P를 만든 후 출력하는 프로그램을 작성하시오. 즉, 첫 번째 문자를 R번 반복하고, 두 번째 문자를 R번 반복하는 식으로 P를 만들면 된다. S에는 QR Code "alphanumeric" 문자만 들어있다.

QR Code "alphanumeric" 문자는 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ\$%*+-./: 이다.

- 입력

첫째 줄에 테스트 케이스의 개수 T(1 ≤ T ≤ 1,000)가 주어진다. 각 테스트 케이스는 반복 횟수 R(1 ≤ R ≤ 8), 문자열 S가 공백으로 구분되어 주어진다. S의 길이는 적어도 1이며, 20글자를 넘지 않는다. 

- 출력

각 테스트 케이스에 대해 P를 출력한다.
```
예제 입력 1 
2
3 ABC
5 /HTP

예제 출력 1 
AAABBBCCC
/////HHHHHTTTTTPPPPP
```

In [154]:
num=int(input())
a = [input().split() for x in range(num)]

for z in a:
    v=''
    for x in z[1]:
        v+=int(z[0])*x
    print(v)

3
3 aeghstf
7 sdfv
2 sbtfb


## 1157
- 문제

알파벳 대소문자로 된 단어가 주어지면, 이 단어에서 가장 많이 사용된 알파벳이 무엇인지 알아내는 프로그램을 작성하시오. 단, 대문자와 소문자를 구분하지 않는다.

- 입력

첫째 줄에 알파벳 대소문자로 이루어진 단어가 주어진다. 주어지는 단어의 길이는 1,000,000을 넘지 않는다.

- 출력

첫째 줄에 이 단어에서 가장 많이 사용된 알파벳을 대문자로 출력한다. 단, 가장 많이 사용된 알파벳이 여러 개 존재하는 경우에는 ?를 출력한다.
```
예제 입력 1 
Mississipi
예제 출력 1 
?
예제 입력 2 
zZa
예제 출력 2 
Z
```

In [291]:
a=input()

A=a.upper()

b=[i for i in set(A)]
d=[A.count(x) for x in b]
m=0
i=''
if d.count(max(d))>1:
    print('?')
else :
    for x in b:
        c=A.count(x)
        if c > m:
            m=c
            i=x
    print(i)

aaajjjbrrrbbhh
?


## 1152
- 문제

영어 대소문자와 띄어쓰기만으로 이루어진 문자열이 주어진다. 이 문자열에는 몇 개의 단어가 있을까? 이를 구하는 프로그램을 작성하시오. 단, 한 단어가 여러 번 등장하면 등장한 횟수만큼 모두 세어야 한다.

- 입력

첫 줄에 영어 대소문자와 띄어쓰기로 이루어진 문자열이 주어진다. 이 문자열의 길이는 1,000,000을 넘지 않는다. 단어는 띄어쓰기 한 개로 구분되며, 공백이 연속해서 나오는 경우는 없다. 또한 문자열의 앞과 뒤에는 공백이 있을 수도 있다.

- 출력

첫째 줄에 단어의 개수를 출력한다.
```
예제 입력 1 
The Curious Case of Benjamin Button
예제 출력 1 
6
```

In [297]:
print(len(input().split()))

The Curious Case of Benjamin Button


6

## 2908
- 문제

상근이의 동생 상수는 수학을 정말 못한다. 상수는 숫자를 읽는데 문제가 있다. 이렇게 수학을 못하는 상수를 위해서 상근이는 수의 크기를 비교하는 문제를 내주었다. 상근이는 세 자리 수 두 개를 칠판에 써주었다. 그 다음에 크기가 큰 수를 말해보라고 했다.

상수는 수를 다른 사람과 다르게 거꾸로 읽는다. 예를 들어, 734과 893을 칠판에 적었다면, 상수는 이 수를 437과 398로 읽는다. 따라서, 상수는 두 수중 큰 수인 437을 큰 수라고 말할 것이다.

두 수가 주어졌을 때, 상수의 대답을 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 상근이가 칠판에 적은 두 수 A와 B가 주어진다. 두 수는 같지 않은 세 자리 수이며, 0이 포함되어 있지 않다.

- 출력

첫째 줄에 상수의 대답을 출력한다.
```
예제 입력 1 
734 893
예제 출력 1 
437
```

In [1]:
a,b=map(list,input().split())

for i in range(2,0,-1):
    if int(a[i]) > int(b[i]):
        print(int(a[2])*100+int(a[1])*10+int(a[0]))
        break
    elif int(b[i]) > int(a[i]):
        print(int(b[2])*100+int(b[1])*10+int(b[0]))
        break

 734 893


437


### 5622

<img src=https://onlinejudgeimages.s3-ap-northeast-1.amazonaws.com/upload/images/dial.png align='right'>

- 문제

상근이의 할머니는 아래 그림과 같이 오래된 다이얼 전화기를 사용한다.

전화를 걸고 싶은 번호가 있다면, 숫자를 하나를 누른 다음에 금속 핀이 있는 곳 까지 시계방향으로 돌려야 한다. 숫자를 하나 누르면 다이얼이 처음 위치로 돌아가고, 다음 숫자를 누르려면 다이얼을 처음 위치에서 다시 돌려야 한다.

숫자 1을 걸려면 총 2초가 필요하다. 1보다 큰 수를 거는데 걸리는 시간은 이보다 더 걸리며, 한 칸 옆에 있는 숫자를 걸기 위해선 1초씩 더 걸린다.

상근이의 할머니는 전화 번호를 각 숫자에 해당하는 문자로 외운다. 즉, 어떤 단어를 걸 때, 각 알파벳에 해당하는 숫자를 걸면 된다. 예를 들어, UNUCIC는 868242와 같다.

할머니가 외운 단어가 주어졌을 때, 이 전화를 걸기 위해서 필요한 시간을 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 알파벳 대문자로 이루어진 단어가 주어진다. 단어는 2글자~15글자로 이루어져 있다.

- 출력

첫째 줄에 다이얼을 걸기 위해서 필요한 시간을 출력한다.
```
예제 입력 1 
UNUCIC
예제 출력 1 
36
```

In [51]:
input1 = 'UNUCIC'
dial=[['ABC'],['DEF'],['GHI'],['JKL'],['MNO'],['PQRS'],['TUV'],['WXYZ']]

In [53]:
num=[]
for inp in inputs:
    i=1
    for a in dial:
        i += 1
        if inp in a[0]:
            num.append(i+1)
print(sum(num))

36


### 2941
- 문제

예전에는 운영체제에서 크로아티아 알파벳을 입력할 수가 없었다. 따라서, 다음과 같이 크로아티아 알파벳을 변경해서 입력했다.


크로아티아 알파벳|변경
```
크로아티아 알파벳	변경
č	c=
ć	c-
dž	dz=
đ	d-
lj	lj
nj	nj
š	s=
ž	z=
```
예를 들어, ljes=njak은 크로아티아 알파벳 6개(lj, e, š, nj, a, k)로 이루어져 있다. 단어가 주어졌을 때, 몇 개의 크로아티아 알파벳으로 이루어져 있는지 출력한다.

dž는 무조건 하나의 알파벳으로 쓰이고, d와 ž가 분리된 것으로 보지 않는다. lj와 nj도 마찬가지이다. 위 목록에 없는 알파벳은 한 글자씩 센다.

- 입력

첫째 줄에 최대 100글자의 단어가 주어진다. 알파벳 소문자와 '-', '='로만 이루어져 있다.

단어는 크로아티아 알파벳으로 이루어져 있다. 문제 설명의 표에 나와있는 알파벳은 변경된 형태로 입력된다.

- 출력

입력으로 주어진 단어가 몇 개의 크로아티아 알파벳으로 이루어져 있는지 출력한다.
```
예제 입력 1 
ljes=njak
예제 출력 1 
6
예제 입력 2 
ddz=z=
예제 출력 2 
3
```

In [127]:
a=input()

b=['c=','c-','dz=','d-','lj','nj','s=','z=']

for asd in b:
    a=a.replace(asd,'a')
print(len(a))

 ljes=njak


6


### 1316
- 문제

그룹 단어란 단어에 존재하는 모든 문자에 대해서, 각 문자가 연속해서 나타나는 경우만을 말한다. 예를 들면, ccazzzzbb는 c, a, z, b가 모두 연속해서 나타나고, kin도 k, i, n이 연속해서 나타나기 때문에 그룹 단어이지만, aabbbccb는 b가 떨어져서 나타나기 때문에 그룹 단어가 아니다.

단어 N개를 입력으로 받아 그룹 단어의 개수를 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 단어의 개수 N이 들어온다. N은 100보다 작거나 같은 자연수이다. 둘째 줄부터 N개의 줄에 단어가 들어온다. 단어는 알파벳 소문자로만 되어있고 중복되지 않으며, 길이는 최대 100이다.

- 출력

첫째 줄에 그룹 단어의 개수를 출력한다.
```
예제 입력 1 
3
happy
new
year
예제 출력 1 
3
예제 입력 2 
4
aba
abab
abcabc
a
예제 출력 2 
1
```

In [234]:
a=int(input())
b=[input() for _ in range(a)]
cnt=len(b)

for bbb in b:
    for i,a in enumerate(bbb):
        if bbb.count(a,i)+bbb.find(a,i)!=bbb.rfind(a)+1:
            cnt -= 1
            break

print(cnt)

 3
 happy
 new
 year


3


### 1712
- 문제

월드전자는 노트북을 제조하고 판매하는 회사이다. 노트북 판매 대수에 상관없이 매년 임대료, 재산세, 보험료, 급여 등 A만원의 고정 비용이 들며, 한 대의 노트북을 생산하는 데에는 재료비와 인건비 등 총 B만원의 가변 비용이 든다고 한다.

예를 들어 A=1,000, B=70이라고 하자. 이 경우 노트북을 한 대 생산하는 데는 총 1,070만원이 들며, 열 대 생산하는 데는 총 1,700만원이 든다.

노트북 가격이 C만원으로 책정되었다고 한다. 일반적으로 생산 대수를 늘려 가다 보면 어느 순간 총 수입(판매비용)이 총 비용(=고정비용+가변비용)보다 많아지게 된다. 최초로 총 수입이 총 비용보다 많아져 이익이 발생하는 지점을 손익분기점(BREAK-EVEN POINT)이라고 한다.

A, B, C가 주어졌을 때, 손익분기점을 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 A, B, C가 빈 칸을 사이에 두고 순서대로 주어진다. A, B, C는 21억 이하의 자연수이다.

- 출력

첫 번째 줄에 손익분기점 즉 최초로 이익이 발생하는 판매량을 출력한다. 손익분기점이 존재하지 않으면 -1을 출력한다.
```
예제 입력 1 
1000 70 170
예제 출력 1 
11
```

In [330]:
a,b,c = map(int,input().split())

if b>=c:
    print(-1)
else :
    print(int((a / (c-b))+1))

 1000 70 170


11


## 2839
- 문제

상근이는 요즘 설탕공장에서 설탕을 배달하고 있다. 상근이는 지금 사탕가게에 설탕을 정확하게 N킬로그램을 배달해야 한다. 설탕공장에서 만드는 설탕은 봉지에 담겨져 있다. 봉지는 3킬로그램 봉지와 5킬로그램 봉지가 있다.

상근이는 귀찮기 때문에, 최대한 적은 봉지를 들고 가려고 한다. 예를 들어, 18킬로그램 설탕을 배달해야 할 때, 3킬로그램 봉지 6개를 가져가도 되지만, 5킬로그램 3개와 3킬로그램 1개를 배달하면, 더 적은 개수의 봉지를 배달할 수 있다.

상근이가 설탕을 정확하게 N킬로그램 배달해야 할 때, 봉지 몇 개를 가져가면 되는지 그 수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 N이 주어진다. (3 ≤ N ≤ 5000)

- 출력

상근이가 배달하는 봉지의 최소 개수를 출력한다. 만약, 정확하게 N킬로그램을 만들 수 없다면 -1을 출력한다.
```
예제 입력 1 
18
예제 출력 1 
4
예제 입력 2 
4
예제 출력 2 
-1
예제 입력 3 
6
예제 출력 3 
2
```

In [679]:
kg=int(input())

a=[]
for i in range((kg//5)+1):
    x1=(kg-5*i)/3
    a.append(i+x1)

x=0
for i in range(len(a)):
    i=len(a)-i-1
    if int(a[i])==a[i]:
        x=int(a[i])
        break

if x==0:
    print(-1)
else :
    print(x)

 18


4


## 2292
<img src=https://www.acmicpc.net/JudgeOnline/upload/201009/3(2).png align=right>

- 문제

위의 그림과 같이 육각형으로 이루어진 벌집이 있다. 그림에서 보는 바와 같이 중앙의 방 1부터 시작해서 이웃하는 방에 돌아가면서 1씩 증가하는 번호를 주소로 매길 수 있다. 숫자 N이 주어졌을 때, 벌집의 중앙 1에서 N번 방까지 최소 개수의 방을 지나서 갈 때 몇 개의 방을 지나가는지(시작과 끝을 포함하여)를 계산하는 프로그램을 작성하시오. 예를 들면, 13까지는 3개, 58까지는 5개를 지난다.

- 입력

첫째 줄에 N(1 ≤ N ≤ 1,000,000,000)이 주어진다.

- 출력

입력으로 주어진 방까지 최소 개수의 방을 지나서 갈 때 몇 개의 방을 지나는지 출력한다.
```
예제 입력 1 
13
예제 출력 1 
3
```

In [804]:
n=int(input())
a=6
b=1
c=0
i=1

while True:
    b += i*a
    if n > b:
        c=i+2
        i+=1
    if n <= b:
        break
        
if n==1:
    print(n)
elif 1<n<=7:
    print(2)
else:
    print(c)

 58


5


### 1193
- 문제

무한히 큰 배열에 다음과 같이 분수들이 적혀있다.
```
1/1	1/2	1/3	1/4	1/5	…
2/1	2/2	2/3	2/4	…	…
3/1	3/2	3/3	…	…	…
4/1	4/2	…	…	…	…
5/1	…	…	…	…	…
…	…	…	…	…	…
```
이와 같이 나열된 분수들을 1/1 -> 1/2 -> 2/1 -> 3/1 -> 2/2 -> … 과 같은 지그재그 순서로 차례대로 1번, 2번, 3번, 4번, 5번, … 분수라고 하자.

X가 주어졌을 때, X번째 분수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 X(1 ≤ X ≤ 10,000,000)가 주어진다.

- 출력

첫째 줄에 분수를 출력한다.
```
예제 입력 1 
14
예제 출력 1 
2/4
```

In [10]:
x=int(input())
i=-1
a=[0]
while True:
    i+=1
    a.append(a[i]+i+1)
    if x <= a[i]:
        if i%2==1:
            up=a[i]-x+1
            down=i-up+1
            print(f'{up}/{down}')
        if i%2==0:
            down=a[i]-x+1
            up=i-down+1
            print(f'{up}/{down}')
        break

 14


2/4


## 2869
- 문제

땅 위에 달팽이가 있다. 이 달팽이는 높이가 V미터인 나무 막대를 올라갈 것이다.

달팽이는 낮에 A미터 올라갈 수 있다. 하지만, 밤에 잠을 자는 동안 B미터 미끄러진다. 또, 정상에 올라간 후에는 미끄러지지 않는다.

달팽이가 나무 막대를 모두 올라가려면, 며칠이 걸리는지 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 세 정수 A, B, V가 공백으로 구분되어서 주어진다. (1 ≤ B < A ≤ V ≤ 1,000,000,000)

- 출력

첫째 줄에 달팽이가 나무 막대를 모두 올라가는데 며칠이 걸리는지 출력한다.
```
예제 입력 1 
2 1 5
예제 출력 1 
4
```

In [165]:
day, nig, v = map(int,input().split())

h=v-day

d=day-nig

if (h/d)%1==0 : 
    print(int((h/d))+1)
else : 
    print(int((h/d))+2)

 2 1 1000000000


999999999


## 10250

- 문제

ACM 호텔 매니저 지우는 손님이 도착하는 대로 빈 방을 배정하고 있다. 고객 설문조사에 따르면 손님들은 호텔 정문으로부터 걸어서 가장 짧은 거리에 있는 방을 선호한다고 한다. 여러분은 지우를 도와 줄 프로그램을 작성하고자 한다. 즉 설문조사 결과 대로 호텔 정문으로부터 걷는 거리가 가장 짧도록 방을 배정하는 프로그램을 작성하고자 한다.

문제를 단순화하기 위해서 호텔은 직사각형 모양이라고 가정하자. 각 층에 W 개의 방이 있는 H 층 건물이라고 가정하자 (1 ≤ H, W ≤ 99). 그리고 엘리베이터는 가장 왼쪽에 있다고 가정하자(그림 1 참고). 이런 형태의 호텔을 H × W 형태 호텔이라고 부른다. 호텔 정문은 일층 엘리베이터 바로 앞에 있는데, 정문에서 엘리베이터까지의 거리는 무시한다. 또 모든 인접한 두 방 사이의 거리는 같은 거리(거리 1)라고 가정하고 호텔의 정면 쪽에만 방이 있다고 가정한다.


<img src=https://www.acmicpc.net/upload/images2/elevator.png>
그림 1. H = 6 이고 W = 12 인 H × W 호텔을 간략하게 나타낸 그림

방 번호는 YXX 나 YYXX 형태인데 여기서 Y 나 YY 는 층 수를 나타내고 XX 는 엘리베이터에서부터 세었을 때의 번호를 나타낸다. 즉, 그림 1 에서 빗금으로 표시한 방은 305 호가 된다.

손님은 엘리베이터를 타고 이동하는 거리는 신경 쓰지 않는다. 다만 걷는 거리가 같을 때에는 아래층의 방을 더 선호한다. 예를 들면 102 호 방보다는 301 호 방을 더 선호하는데, 102 호는 거리 2 만큼 걸어야 하지만 301 호는 거리 1 만큼만 걸으면 되기 때문이다. 같은 이유로 102 호보다 2101 호를 더 선호한다.

여러분이 작성할 프로그램은 초기에 모든 방이 비어있다고 가정하에 이 정책에 따라 N 번째로 도착한 손님에게 배정될 방 번호를 계산하는 프로그램이다. 첫 번째 손님은 101 호, 두 번째 손님은 201 호 등과 같이 배정한다. 그림 1 의 경우를 예로 들면, H = 6이므로 10 번째 손님은 402 호에 배정해야 한다.

- 입력

프로그램은 표준 입력에서 입력 데이터를 받는다. 프로그램의 입력은 T 개의 테스트 데이터로 이루어져 있는데 T 는 입력의 맨 첫 줄에 주어진다. 각 테스트 데이터는 한 행으로서 H, W, N, 세 정수를 포함하고 있으며 각각 호텔의 층 수, 각 층의 방 수, 몇 번째 손님인지를 나타낸다(1 ≤ H, W ≤ 99, 1 ≤ N ≤ H × W). 

- 출력

프로그램은 표준 출력에 출력한다. 각 테스트 데이터마다 정확히 한 행을 출력하는데, 내용은 N 번째 손님에게 배정되어야 하는 방 번호를 출력한다.
```
예제 입력 1 
2
6 12 10
30 50 72
예제 출력 1 
402
1203
```

In [325]:
T=int(input())
aa = [list(map(int,input().split())) for _ in range(T)]

for height, width, n in aa:
    if n%height!=0:
        print((n%height)*100+(int(n/height)+1))
    elif n%height==0:
        print(int(height*100+n/height))

## 2775
- 문제

평소 반상회에 참석하는 것을 좋아하는 주희는 이번 기회에 부녀회장이 되고 싶어 각 층의 사람들을 불러 모아 반상회를 주최하려고 한다.

이 아파트에 거주를 하려면 조건이 있는데, “a층의 b호에 살려면 자신의 아래(a-1)층의 1호부터 b호까지 사람들의 수의 합만큼 사람들을 데려와 살아야 한다” 는 계약 조항을 꼭 지키고 들어와야 한다.

아파트에 비어있는 집은 없고 모든 거주민들이 이 계약 조건을 지키고 왔다고 가정했을 때, 주어지는 양의 정수 k와 n에 대해 k층에 n호에는 몇 명이 살고 있는지 출력하라. 단, 아파트에는 0층부터 있고 각층에는 1호부터 있으며, 0층의 i호에는 i명이 산다.

- 입력

첫 번째 줄에 Test case의 수 T가 주어진다. 그리고 각각의 케이스마다 입력으로 첫 번째 줄에 정수 k, 두 번째 줄에 정수 n이 주어진다. (1 <= k <= 14, 1 <= n <= 14)

- 출력

각각의 Test case에 대해서 해당 집에 거주민 수를 출력하라.
```
예제 입력 1 
2
1
3
2
3
예제 출력 1 
6
10
```

In [711]:
def room(k,n):
    res=0
    if n in pre[k]:
        return pre[k][n]
    else:
        for i in range(1,n+1):
            temp = room(k-1,i)
            pre[k-1][i]=temp
            res += temp
        return res

pri=[]
T=int(input())

for _ in range(T):
    k=int(input())
    n=int(input())
    
    pre={}
    for i in range(k+1):
        pre[i]={}
    for j in range(1,n+1):
        pre[0][j]=j
    pri.append(room(k,n))

for i in range(T):
    print(pri[i])

 1
 3
 5


70


## 1011
- 문제

우현이는 어린 시절, 지구 외의 다른 행성에서도 인류들이 살아갈 수 있는 미래가 오리라 믿었다. 그리고 그가 지구라는 세상에 발을 내려 놓은 지 23년이 지난 지금, 세계 최연소 ASNA 우주 비행사가 되어 새로운 세계에 발을 내려 놓는 영광의 순간을 기다리고 있다.

그가 탑승하게 될 우주선은 Alpha Centauri라는 새로운 인류의 보금자리를 개척하기 위한 대규모 생활 유지 시스템을 탑재하고 있기 때문에, 그 크기와 질량이 엄청난 이유로 최신기술력을 총 동원하여 개발한 공간이동 장치를 탑재하였다. 하지만 이 공간이동 장치는 이동 거리를 급격하게 늘릴 경우 기계에 심각한 결함이 발생하는 단점이 있어서, 이전 작동시기에 k광년을 이동하였을 때는 k-1 , k 혹은 k+1 광년만을 다시 이동할 수 있다. 예를 들어, 이 장치를 처음 작동시킬 경우 -1 , 0 , 1 광년을 이론상 이동할 수 있으나 사실상 음수 혹은 0 거리만큼의 이동은 의미가 없으므로 1 광년을 이동할 수 있으며, 그 다음에는 0 , 1 , 2 광년을 이동할 수 있는 것이다. ( 여기서 다시 2광년을 이동한다면 다음 시기엔 1, 2, 3 광년을 이동할 수 있다. )

<img src=https://www.acmicpc.net/upload/201003/rlaehdgur.JPG>

김우현은 공간이동 장치 작동시의 에너지 소모가 크다는 점을 잘 알고 있기 때문에 x지점에서 y지점을 향해 최소한의 작동 횟수로 이동하려 한다. 하지만 y지점에 도착해서도 공간 이동장치의 안전성을 위하여 y지점에 도착하기 바로 직전의 이동거리는 반드시 1광년으로 하려 한다.

김우현을 위해 x지점부터 정확히 y지점으로 이동하는데 필요한 공간 이동 장치 작동 횟수의 최솟값을 구하는 프로그램을 작성하라.

- 입력

입력의 첫 줄에는 테스트케이스의 개수 T가 주어진다. 각각의 테스트 케이스에 대해 현재 위치 x 와 목표 위치 y 가 정수로 주어지며, x는 항상 y보다 작은 값을 갖는다. ( 0 ≤ x < y < 231)

- 출력

각 테스트 케이스에 대해 x지점으로부터 y지점까지 정확히 도달하는데 필요한 최소한의 공간이동 장치 작동 회수를 출력한다.
```
예제 입력 1 
3
0 3
1 5
45 50
예제 출력 1 
3
3
4
```

안해


## 1978
- 문제

주어진 수 N개 중에서 소수가 몇 개인지 찾아서 출력하는 프로그램을 작성하시오.

- 입력

첫 줄에 수의 개수 N이 주어진다. N은 100이하이다. 다음으로 N개의 수가 주어지는데 수는 1,000 이하의 자연수이다.

- 출력

주어진 수들 중 소수의 개수를 출력한다.
```
예제 입력 1 
4
1 3 5 7
예제 출력 1 
3
```

In [133]:
q=input()
num = list(map(int, input().split()))
cnt = len(num)

for a in num:
    if a==1:
        cnt -=1
    else :
        for i in range(2,a):
            if a%i==0:
                cnt -= 1
                break
                
print(cnt)

 3
 1 3 5 7 9


3


## 2581
- 문제

자연수 M과 N이 주어질 때 M이상 N이하의 자연수 중 소수인 것을 모두 골라 이들 소수의 합과 최솟값을 찾는 프로그램을 작성하시오.

예를 들어 M=60, N=100인 경우 60이상 100이하의 자연수 중 소수는 61, 67, 71, 73, 79, 83, 89, 97 총 8개가 있으므로, 이들 소수의 합은 620이고, 최솟값은 61이 된다.

- 입력

입력의 첫째 줄에 M이, 둘째 줄에 N이 주어진다.

M과 N은 10,000이하의 자연수이며, M은 N보다 작거나 같다.

- 출력

M이상 N이하의 자연수 중 소수인 것을 모두 찾아 첫째 줄에 그 합을, 둘째 줄에 그 중 최솟값을 출력한다. 

단, M이상 N이하의 자연수 중 소수가 없을 경우는 첫째 줄에 -1을 출력한다.
```
예제 입력 1 
60
100
예제 출력 1 
620
61
예제 입력 2 
64
65
예제 출력 2 
-1
```

In [283]:
m=int(input())
n=int(input())
nums=[i for i in range(m,n+1)]
list1=[i for i in range(m,n+1)]
# print(nums)
# print(list1)

for a in nums:
#     print(a,'시작')
    if a==1:
#         print(a, '삭제')
        list1.remove(a)
    else :
#         print(a)
        for i in range(2,a):
#             print(a,i,a%i)
            if a%i==0:
#                 print(a,'삭제')
                list1.remove(a)
#                 print(nums)
#                 print(list1)
                break
                
# print(list1)
if len(list1)==0:
    print(-1)
else :
    print(sum(list1))
    print(list1[0])

 64
 65


-1


## 1929
- 문제

M이상 N이하의 소수를 모두 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 자연수 M과 N이 빈 칸을 사이에 두고 주어진다. (1 ≤ M ≤ N ≤ 1,000,000)

- 출력

한 줄에 하나씩, 증가하는 순서대로 소수를 출력한다.
```
예제 입력 1 
3 16
예제 출력 1 
3
5
7
11
13
```

In [130]:
def issosu(v):
    for i in range(2,int(v**0.5)+1):   # 제곱근까지만 확인하면 됨. 이게 포인트.
        if v%i==0:
            return 0
    return 1*(v!=1)

a,b = map(int, input().split())
for i in range(a, b+1):
    if issosu(i)==1:
        print(i)

 1 100


2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97


## 4948
- 문제

베르트랑 공준은 임의의 자연수 n에 대하여, n보다 크고, 2n보다 작거나 같은 소수는 적어도 하나 존재한다는 내용을 담고 있다.
이 명제는 조제프 베르트랑이 1845년에 추측했고, 파프누티 체비쇼프가 1850년에 증명했다.
예를 들어, 10보다 크고, 20보다 작거나 같은 소수는 4개가 있다. (11, 13, 17, 19) 또, 14보다 크고, 28보다 작거나 같은 소수는 3개가 있다. (17,19, 23)
n이 주어졌을 때, n보다 크고, 2n보다 작거나 같은 소수의 개수를 구하는 프로그램을 작성하시오. 

- 입력

입력은 여러 개의 테스트 케이스로 이루어져 있다. 각 케이스는 n을 포함하며, 한 줄로 이루어져 있다. (n ≤ 123456)
입력의 마지막에는 0이 주어진다.

- 출력

각 테스트 케이스에 대해서, n보다 크고, 2n보다 작거나 같은 소수의 개수를 출력한다.
```
예제 입력 1 
1
10
13
100
1000
10000
100000
0
예제 출력 1 
1
4
3
21
135
1033
8392
```

In [44]:
import sys

def issosu(v):
    for i in range(2,int(v**0.5)+1):   # 제곱근까지만 확인하면 됨. 이게 포인트.
        if v%i==0:
            return 0
    return 1*(v!=1)

primelist=[0 for _ in range(123456*2+1)]

for i in range(len(primelist)):
    if issosu(i)==1:
        primelist[i]=1

num=[]
while True:
    a=int(sys.stdin.readline())
    num.append(a)
    if a==0:
        break

for n in num:
    if n!=0:
        print(primelist[n+1:2*n+1].count(1))

1
4
3
21
135
1033
8392


## 9020
- 문제

1보다 큰 자연수 중에서  1과 자기 자신을 제외한 약수가 없는 자연수를 소수라고 한다. 예를 들어, 5는 1과 5를 제외한 약수가 없기 때문에 소수이다. 하지만, 6은 6 = 2 × 3 이기 때문에 소수가 아니다.
골드바흐의 추측은 유명한 정수론의 미해결 문제로, 2보다 큰 모든 짝수는 두 소수의 합으로 나타낼 수 있다는 것이다. 이러한 수를 골드바흐 수라고 한다. 또, 짝수를 두 소수의 합으로 나타내는 표현을 그 수의 골드바흐 파티션이라고 한다. 예를 들면, 4 = 2 + 2, 6 = 3 + 3, 8 = 3 + 5, 10 = 5 + 5, 12 = 5 + 7, 14 = 3 + 11, 14 = 7 + 7이다. 10000보다 작거나 같은 모든 짝수 n에 대한 골드바흐 파티션은 존재한다.
2보다 큰 짝수 n이 주어졌을 때, n의 골드바흐 파티션을 출력하는 프로그램을 작성하시오. 만약 가능한 n의 골드바흐 파티션이 여러 가지인 경우에는 두 소수의 차이가 가장 작은 것을 출력한다.

- 입력

첫째 줄에 테스트 케이스의 개수 T가 주어진다. 각 테스트 케이스는 한 줄로 이루어져 있고 짝수 n이 주어진다. (4 ≤ n ≤ 10,000)

- 출력

각 테스트 케이스에 대해서 주어진 n의 골드바흐 파티션을 출력한다. 출력하는 소수는 작은 것부터 먼저 출력하며, 공백으로 구분한다.
```
예제 입력 1 
3
8
10
16
예제 출력 1 
3 5
5 5
5 11
```

In [26]:
def issosu(v):
    for i in range(2,int(v**0.5)+1):   
        if v%i==0:
            return 0
    return 1*(v!=1)

nums=[]
a=int(input())
for _ in range(a):
    nums.append(int(input()))


for num in nums:
    for i in range(int(num/2),num):
        if issosu(i)==1 and issosu(num-i)==1:
            print(num-i,i)
            break

 3
 8
 10
 16


3 5
5 5
5 11


## 1085
- 문제

한수는 지금 (x, y)에 있다. 직사각형의 왼쪽 아래 꼭짓점은 (0, 0)에 있고, 오른쪽 위 꼭짓점은 (w, h)에 있다. 직사각형의 경계선까지 가는 거리의 최솟값을 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 x y w h가 주어진다. w와 h는 1,000보다 작거나 같은 자연수이고, x는 1보다 크거나 같고, w-1보다 작거나 같은 자연수이고, y는 1보다 크거나 같고, h-1보다 작거나 같은 자연수이다.

- 출력

첫째 줄에 문제의 정답을 출력한다.
```
예제 입력 1 
6 2 10 3
예제 출력 1 
1
```

In [44]:
n=list(map(int, input().split()))

dx=n[2]-n[0]
dy=n[3]-n[1]

n=[dx,dy,n[0],n[1]]
print(min(n))

 2 3 100 50


2


## 3009
- 문제

세 점이 주어졌을 때, 축에 평행한 직사각형을 만들기 위해서 필요한 네 번째 점을 찾는 프로그램을 작성하시오.

- 입력

세 점의 좌표가 한 줄에 하나씩 주어진다. 좌표는 1보다 크거나 같고, 1000보다 작거나 같은 정수이다.

- 출력

직사각형의 네 번째 점의 좌표를 출력한다.
```
예제 입력 1 
30 20
10 10
10 20
예제 출력 1 
30 10
```

In [64]:
x=[]
y=[]
for _ in range(3):
    a,b = map(int,input().split())
    x.append(a)
    y.append(b)

dot=[0,0]

for i in range(3):
    if x.count(x[i])==1:
        dot[0]=x[i]
    if y.count(y[i])==1:
        dot[1]=y[i]

print(dot[0],dot[1])

## 4153
- 문제

과거 이집트인들은 각 변들의 길이가 3, 4, 5인 삼각형이 직각 삼각형인것을 알아냈다. 주어진 세변의 길이로 삼각형이 직각인지 아닌지 구분하시오.

- 입력

입력은 여러개의 테스트케이스로 주어지며 마지막줄에는 0 0 0이 입력된다. 각 테스트케이스는 모두 30,000보다 작은 양의 정수로 주어지며, 각 입력은 변의 길이를 의미한다.

- 출력

각 입력에 대해 직각 삼각형이 맞다면 "right", 아니라면 "wrong"을 출력한다.
```
예제 입력 1 
6 8 10
25 52 60
5 12 13
0 0 0
예제 출력 1 
right
wrong
right
```

In [101]:
nums=[]
num = list(map(int, input().split()))
while 0 not in num:
    nums.append(num)
    num = list(map(int, input().split()))

for num in nums:
    num.sort()
    if num[0]**2+num[1]**2==num[2]**2: print('right')
    else : print('wrong')

## 3053
- 문제

19세기 독일 수학자 헤르만 민코프스키는 비유클리드 기하학 중 택시 기하학을 고안했다.

택시 기하학에서 두 점 T1(x1,y1), T2(x2,y2) 사이의 거리는 다음과 같이 구할 수 있다.

D(T1,T2) = |x1-x2| + |y1-y2|

두 점 사이의 거리를 제외한 나머지 정의는 유클리드 기하학에서의 정의와 같다.

따라서 택시 기하학에서 원의 정의는 유클리드 기하학에서 원의 정의와 같다.

원: 평면 상의 어떤 점에서 거리가 일정한 점들의 집합

반지름 R이 주어졌을 때, 유클리드 기하학에서 원의 넓이와, 택시 기하학에서 원의 넓이를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 반지름 R이 주어진다. R은 10,000보다 작거나 같은 자연수이다.

- 출력

첫째 줄에는 유클리드 기하학에서 반지름이 R인 원의 넓이를, 둘째 줄에는 택시 기하학에서 반지름이 R인 원의 넓이를 출력한다. 정답과의 오차는 0.0001까지 허용한다.
```
예제 입력 1 
1
예제 출력 1 
3.141593
2.000000
```

In [115]:
import math

r=int(input())

print(math.pi*r**2)

print(2*r**2)

## 1002
- 문제

조규현과 백승환은 터렛에 근무하는 직원이다. 하지만 워낙 존재감이 없어서 인구수는 차지하지 않는다. 다음은 조규현과 백승환의 사진이다.
이석원은 조규현과 백승환에게 상대편 마린(류재명)의 위치를 계산하라는 명령을 내렸다. 조규현과 백승환은 각각 자신의 터렛 위치에서 현재 적까지의 거리를 계산했다.
조규현의 좌표 (x1, y1)와 백승환의 좌표 (x2, y2)가 주어지고, 조규현이 계산한 류재명과의 거리 r1과 백승환이 계산한 류재명과의 거리 r2가 주어졌을 때, 류재명이 있을 수 있는 좌표의 수를 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 테스트 케이스의 개수 T가 주어진다. 각 테스트 케이스는 다음과 같이 이루어져 있다.
한 줄에 x1, y1, r1, x2, y2, r2가 주어진다. x1, y1, x2, y2는 -10,000보다 크거나 같고, 10,000보다 작거나 같은 정수이고, r1, r2는 10,000보다 작거나 같은 자연수이다.

- 출력

각 테스트 케이스마다 류재명이 있을 수 있는 위치의 수를 출력한다. 만약 류재명이 있을 수 있는 위치의 개수가 무한대일 경우에는 -1을 출력한다.
```
예제 입력 1 
3
0 0 13 40 0 37
0 0 3 0 7 4
1 1 1 1 1 5
예제 출력 1 
2
1
0
```

In [172]:
p=[0,0,13, 40, 0, 37]

In [177]:
p=[0, 0, 3, 0, 7, 4]

In [165]:
p=[1, 1, 1, 1, 1, 5]

In [197]:
ps=[]
T = int(input())
for _ in range(T):
    ps.append(list(map(int, input().split())))

 1
 0 0 6 0 1 4


In [198]:
for p in ps:
    dx=abs(p[3]-p[0])
    dy=abs(p[4]-p[1])
    d=(dx**2+dy**2)**0.5   #원의 중심간의 거리
    
    r1=p[2]
    r2=p[5]
    r=r1+r2      #반지름들의 합
    
    if r1>d or r2>d : # 원이 다른 한 원을 포함할때
        if r1>r2: #r1이 r2를 포함할때.
            if r1 > d + r2:print(0) #접점 0
            elif r1 == d+r2:print(1) #접점 1
            elif d==0 and r1==r2 : print(-1)
            else : print(2) # 접점 2
        elif r2>r1: #r2가 r1을 포함할때.
            if r2 > d + r1:print(0) #접점 0
            elif r2 == d+r1:print(1) #접점 1
            elif d==0 and r1==r2 : print(-1)
            else : print(2) # 접점 2
    
    
#     if d==0: #원이 겹칠때.
#         if p[2]==p[5]: print(-1)  #반지름이 같을경우
#         else : print(0)  # 반지름이 다를경우
#     elif  :
        

        
    else :  
        if d == r : print(1) #접점이 하나. 외접
        elif d<r : print(2) #접점이 2개
        elif d>r : print(0)  # 접점 없음.
            
# 나중에 다시

0


## 10872
- 문제

0보다 크거나 같은 정수 N이 주어진다. 이때, N!을 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 정수 N(0 ≤ N ≤ 12)가 주어진다.

- 출력

첫째 줄에 N!을 출력한다.
```
예제 입력 1 
10
예제 출력 1 
3628800
```

In [212]:
def fac(n):
    if n==1 or n==0:
        return 1
    return n*fac(n-1)

print(fac(int(input())))

 10


3628800


## 10870
- 문제

피보나치 수는 0과 1로 시작한다. 0번째 피보나치 수는 0이고, 1번째 피보나치 수는 1이다. 그 다음 2번째 부터는 바로 앞 두 피보나치 수의 합이 된다.

이를 식으로 써보면 Fn = Fn-1 + Fn-2 (n>=2)가 된다.

n=17일때 까지 피보나치 수를 써보면 다음과 같다.

0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597

n이 주어졌을 때, n번째 피보나치 수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 n이 주어진다. n은 20보다 작거나 같은 자연수 또는 0이다.

- 출력

첫째 줄에 n번째 피보나치 수를 출력한다.
```
예제 입력 1 
10
예제 출력 1 
55
```

In [227]:
def fib(n):
    if n==1 or n==2:
        return 1
    elif n==0:
        return 0
    return fib(n-1)+fib(n-2)

print(fib(int(input())))

## 2447
- 문제

예제를 보고 규칙을 유추한 뒤에 별을 찍어 보세요.

- 입력

첫째 줄에 N이 주어진다. N은 항상 3의 제곱꼴인 수이다. (3, 9, 27, ...) (N=3k, 1 ≤ k < 8)

- 출력

첫째 줄부터 N번째 줄까지 별을 출력한다.
```
예제 입력 1 
27
예제 출력 1 
***************************
* ** ** ** ** ** ** ** ** *
***************************
***   ******   ******   ***
* *   * ** *   * ** *   * *
***   ******   ******   ***
***************************
* ** ** ** ** ** ** ** ** *
***************************
*********         *********
* ** ** *         * ** ** *
*********         *********
***   ***         ***   ***
* *   * *         * *   * *
***   ***         ***   ***
*********         *********
* ** ** *         * ** ** *
*********         *********
***************************
* ** ** ** ** ** ** ** ** *
***************************
***   ******   ******   ***
* *   * ** *   * ** *   * *
***   ******   ******   ***
***************************
* ** ** ** ** ** ** ** ** *
***************************
```

In [90]:
# 모르겠다 나중에 다시. https://claude-u.tistory.com/146

## 11729
- 문제

세 개의 장대가 있고 첫 번째 장대에는 반경이 서로 다른 n개의 원판이 쌓여 있다. 각 원판은 반경이 큰 순서대로 쌓여있다. 이제 수도승들이 다음 규칙에 따라 첫 번째 장대에서 세 번째 장대로 옮기려 한다.
한 번에 한 개의 원판만을 다른 탑으로 옮길 수 있다.
쌓아 놓은 원판은 항상 위의 것이 아래의 것보다 작아야 한다.
이 작업을 수행하는데 필요한 이동 순서를 출력하는 프로그램을 작성하라. 단, 이동 횟수는 최소가 되어야 한다.

- 입력

첫째 줄에 첫 번째 장대에 쌓인 원판의 개수 N (1 ≤ N ≤ 20)이 주어진다.
- 출력

첫째 줄에 옮긴 횟수 K를 출력한다.
두 번째 줄부터 수행 과정을 출력한다. 두 번째 줄부터 K개의 줄에 걸쳐 두 정수 A B를 빈칸을 사이에 두고 출력하는데, 이는 A번째 탑의 가장 위에 있는 원판을 B번째 탑의 가장 위로 옮긴다는 뜻이다.
```
예제 입력 1 
3
예제 출력 1 
7
1 3
1 2
3 2
1 3
2 1
2 3
1 3
```

In [1]:
cnt=0
fromlist=[]
tolist=[]
def hanoi(n,from1, by, to):
    global cnt, fromlist, tolist
    if n==1:
#         print(f'{cnt}11 : {n}이 {from1}에서 {to}로 이동했습니다.')
        fromlist.append(from1)
        tolist.append(to)
        cnt += 1
        
    else :
        hanoi(n-1,from1, to, by)
#         print(f'{cnt}22 : {n}이 {from1}에서 {to}로 이동했습니다.')
        fromlist.append(from1)
        tolist.append(to)
        cnt += 1
        hanoi(n-1,by, from1, to)

n=int(input())

hanoi(n, '1', '2', '3')

print(cnt)
for i in range(len(fromlist)):
    print(fromlist[i],tolist[i])

 3


7
1 3
1 2
3 2
1 3
2 1
2 3
1 3


## 2798
- 문제

카지노에서 제일 인기 있는 게임 블랙잭의 규칙은 상당히 쉽다. 카드의 합이 21을 넘지 않는 한도 내에서, 카드의 합을 최대한 크게 만드는 게임이다. 블랙잭은 카지노마다 다양한 규정이 있다.
한국 최고의 블랙잭 고수 김정인은 새로운 블랙잭 규칙을 만들어 상근, 창영이와 게임하려고 한다.
김정인 버젼의 블랙잭에서 각 카드에는 양의 정수가 쓰여 있다. 그 다음, 딜러는 N장의 카드를 모두 숫자가 보이도록 바닥에 놓는다. 그런 후에 딜러는 숫자 M을 크게 외친다.
이제 플레이어는 제한된 시간 안에 N장의 카드 중에서 3장의 카드를 골라야 한다. 블랙잭 변형 게임이기 때문에, 플레이어가 고른 카드의 합은 M을 넘지 않으면서 M과 최대한 가깝게 만들어야 한다.
N장의 카드에 써져 있는 숫자가 주어졌을 때, M을 넘지 않으면서 M에 최대한 가까운 카드 3장의 합을 구해 출력하시오.

- 입력

첫째 줄에 카드의 개수 N(3 ≤ N ≤ 100)과 M(10 ≤ M ≤ 300,000)이 주어진다. 둘째 줄에는 카드에 쓰여 있는 수가 주어지며, 이 값은 100,000을 넘지 않는다.
합이 M을 넘지 않는 카드 3장을 찾을 수 있는 경우만 입력으로 주어진다.

- 출력

첫째 줄에 M을 넘지 않으면서 M에 최대한 가까운 카드 3장의 합을 출력한다.
```
예제 입력 1 
5 21
5 6 7 8 9
예제 출력 1 
21
```

In [173]:
_,bj = map(int,input().split())
card = list(map(int,input().split()))
card.sort(reverse=True)

sumdeck = []
isbj = False
for i in range(len(card)-2):
    deck = []
    deck.append(card[i])
    for j in range(i+1,len(card)):
        deck.append(card[j])
        for k in range(j+1, len(card)):
            deck.append(card[k])
            a=sum(deck)
            sumdeck.append(a)
            if sumdeck[0]<bj or a==bj:
                isbj=True
                break
            deck.remove(card[k])
        deck.remove(card[j])
        if isbj == True:
                break
    if isbj == True:
                break

while True:
    if bj in sumdeck:
        print(bj)
        break
    else:
        bj -= 1

 1 100
 3 5 7 9 15 35 24 55 


99


## 2231
- 문제

어떤 자연수 N이 있을 때, 그 자연수 N의 분해합은 N과 N을 이루는 각 자리수의 합을 의미한다. 어떤 자연수 M의 분해합이 N인 경우, M을 N의 생성자라 한다. 예를 들어, 245의 분해합은 256(=245+2+4+5)이 된다. 따라서 245는 256의 생성자가 된다. 물론, 어떤 자연수의 경우에는 생성자가 없을 수도 있다. 반대로, 생성자가 여러 개인 자연수도 있을 수 있다.

자연수 N이 주어졌을 때, N의 가장 작은 생성자를 구해내는 프로그램을 작성하시오.

- 입력

첫째 줄에 자연수 N(1 ≤ N ≤ 1,000,000)이 주어진다.

- 출력

첫째 줄에 답을 출력한다. 생성자가 없는 경우에는 0을 출력한다.
```
예제 입력 1 
216
예제 출력 1 
198
```

In [61]:
num=int(input())

gen=0
isgen = False

for _ in range(num):
    gen += 1
    gencopy = gen
    strgen = str(gen)
    for n in strgen:
        gencopy+=int(n)
    if gencopy == num:
        print(gen)
        isgen=True
        break
if isgen==False:
    print(0)

 216


198


## 7568
- 문제

우리는 사람의 덩치를 키와 몸무게, 이 두 개의 값으로 표현하여 그 등수를 매겨보려고 한다. 어떤 사람의 몸무게가 x kg이고 키가 y cm라면 이 사람의 덩치는 (x,y)로 표시된다. 두 사람 A 와 B의 덩치가 각각 (x,y), (p,q)라고 할 때 x>p 그리고 y>q 이라면 우리는 A의 덩치가 B의 덩치보다 "더 크다"고 말한다. 예를 들어 어떤 A, B 두 사람의 덩치가 각각 (56,177), (45,165) 라고 한다면 A의 덩치가 B보다 큰 셈이 된다. 그런데 서로 다른 덩치끼리 크기를 정할 수 없는 경우도 있다. 예를 들어 두 사람 C와 D의 덩치가 각각 (45, 181), (55,173)이라면 몸무게는 D가 C보다 더 무겁고, 키는 C가 더 크므로, "덩치"로만 볼 때 C와 D는 누구도 상대방보다 더 크다고 말할 수 없다.

N명의 집단에서 각 사람의 덩치 등수는 자신보다 더 "큰 덩치"의 사람의 수로 정해진다. 만일 자신보다 더 큰 덩치의 사람이 k명이라면 그 사람의 덩치 등수는 k+1이 된다. 이렇게 등수를 결정하면 같은 덩치 등수를 가진 사람은 여러 명도 가능하다. 아래는 5명으로 이루어진 집단에서 각 사람의 덩치와 그 등수가 표시된 표이다.
```
이름	<몸무게, 키>	덩치 등수
A	   <55, 185>	   2
B	   <58, 183>	   2
C	   <88, 186>	   1
D	   <60, 175>	   2
E	   <46, 155>	   5
```
위 표에서 C보다 더 큰 덩치의 사람이 없으므로 C는 1등이 된다. 그리고 A, B, D 각각의 덩치보다 큰 사람은 C뿐이므로 이들은 모두 2등이 된다. 그리고 E보다 큰 덩치는 A, B, C, D 이렇게 4명이므로 E의 덩치는 5등이 된다. 위 경우에 3등과 4등은 존재하지 않는다. 여러분은 학생 N명의 몸무게와 키가 담긴 입력을 읽어서 각 사람의 덩치 등수를 계산하여 출력해야 한다.

- 입력

첫 줄에는 전체 사람의 수 N이 주어진다. 그리고 이어지는 N개의 줄에는 각 사람의 몸무게와 키를 나타내는 양의 정수 x와 y가 하나의 공백을 두고 각각 나타난다. 단, 2 ≤ N ≤ 50, 10 ≤ x,y ≤ 200 이다.

- 출력

여러분은 입력에 나열된 사람의 덩치 등수를 구해서 그 순서대로 첫 줄에 출력해야 한다. 단, 각 덩치 등수는 공백문자로 분리되어야 한다.
```
예제 입력 1 
5
55 185
58 183
88 186
60 175
46 155
예제 출력 1 
2 2 1 2 5
```

In [23]:
n = int(input())
kg=[]
hei=[]
for _ in range(n):
    a, b =map(int, input().split())
    kg.append(a)
    hei.append(b)

res = ''
for a,b in zip(kg,hei):
    bigger = 1
    for i in range(len(kg)):
        if kg[i]>a and hei[i]>b:
            bigger += 1
    res += str(bigger)+' '
print(res)

 5
 55 185
 58 183
 88 186
 60 175
 46 155


2 2 1 2 5 


## 1018
- 문제

지민이는 자신의 저택에서 MN개의 단위 정사각형으로 나누어져 있는 M*N 크기의 보드를 찾았다. 어떤 정사각형은 검은색으로 칠해져 있고, 나머지는 흰색으로 칠해져 있다. 지민이는 이 보드를 잘라서 8*8 크기의 체스판으로 만들려고 한다.

체스판은 검은색과 흰색이 번갈아서 칠해져 있어야 한다. 구체적으로, 각 칸이 검은색과 흰색 중 하나로 색칠되어 있고, 변을 공유하는 두 개의 사각형은 다른 색으로 칠해져 있어야 한다. 따라서 이 정의를 따르면 체스판을 색칠하는 경우는 두 가지뿐이다. 하나는 맨 왼쪽 위 칸이 흰색인 경우, 하나는 검은색인 경우이다.

보드가 체스판처럼 칠해져 있다는 보장이 없어서, 지민이는 8*8 크기의 체스판으로 잘라낸 후에 몇 개의 정사각형을 다시 칠해야겠다고 생각했다. 당연히 8*8 크기는 아무데서나 골라도 된다. 지민이가 다시 칠해야 하는 정사각형의 최소 개수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 N과 M이 주어진다. N과 M은 8보다 크거나 같고, 50보다 작거나 같은 자연수이다. 둘째 줄부터 N개의 줄에는 보드의 각 행의 상태가 주어진다. B는 검은색이며, W는 흰색이다.

- 출력

첫째 줄에 지민이가 다시 칠해야 하는 정사각형 개수의 최솟값을 출력한다.
```
예제 입력 1 
8 8
WBWBWBWB
BWBWBWBW
WBWBWBWB
BWBBBWBW
WBWBWBWB
BWBWBWBW
WBWBWBWB
BWBWBWBW
예제 출력 1 
1
예제 입력 2 
10 13
BBBBBBBBWBWBW
BBBBBBBBBWBWB
BBBBBBBBWBWBW
BBBBBBBBBWBWB
BBBBBBBBWBWBW
BBBBBBBBBWBWB
BBBBBBBBWBWBW
BBBBBBBBBWBWB
WWWWWWWWWWBWB
WWWWWWWWWWBWB
예제 출력 2 
12
```

In [111]:
row, col = map(int, input().split())

chess=''
for _ in range(row):
    chess+=input()

fix=[]
nextrow = 0
for _ in range(row-8+1):
    for i in range(col-8+1):       
        chess_cut=chess[i+nextrow:]  
        nextrow1 = 0
        cor=0
        for i in range(8):
            for j in range(8):    
                if (j+i)%2==0:
                    if chess_cut[j+nextrow1] is 'W':
                        cor += 1
                elif (j+i)%2==1:
                    if chess_cut[j+nextrow1] is 'B':
                        cor += 1
            nextrow1 += col
        fix.append(cor)
        fix.append(64-cor)
    nextrow += col
print(min(fix))

 10 13
 BBBBBBBBWBWBW
 BBBBBBBBBWBWB
 BBBBBBBBWBWBW
 BBBBBBBBBWBWB
 BBBBBBBBWBWBW
 BBBBBBBBBWBWB
 BBBBBBBBWBWBW
 BBBBBBBBBWBWB
 WWWWWWWWWWBWB
 WWWWWWWWWWBWB


12

## 1436
- 문제

666은 종말을 나타내는 숫자라고 한다. 따라서, 많은 블록버스터 영화에서는 666이 들어간 제목을 많이 사용한다. 영화감독 숌은 세상의 종말 이라는 시리즈 영화의 감독이다. 조지 루카스는 스타워즈를 만들 때, 스타워즈 1, 스타워즈 2, 스타워즈 3, 스타워즈 4, 스타워즈 5, 스타워즈 6과 같이 이름을 지었고, 피터 잭슨은 반지의 제왕을 만들 때, 반지의 제왕 1, 반지의 제왕 2, 반지의 제왕 3과 같이 영화 제목을 지었다.
하지만 숌은 자신이 조지 루카스와 피터 잭슨을 뛰어넘는다는 것을 보여주기 위해서 영화 제목을 좀 다르게 만들기로 했다.
종말의 숫자란 어떤 수에 6이 적어도 3개이상 연속으로 들어가는 수를 말한다. 제일 작은 종말의 숫자는 666이고, 그 다음으로 큰 수는 1666, 2666, 3666, .... 과 같다.
따라서, 숌은 첫 번째 영화의 제목은 세상의 종말 666, 두 번째 영화의 제목은 세상의 종말 1666 이렇게 이름을 지을 것이다. 일반화해서 생각하면, N번째 영화의 제목은 세상의 종말 (N번째로 작은 종말의 숫자) 와 같다.
숌이 만든 N번째 영화의 제목에 들어간 숫자를 출력하는 프로그램을 작성하시오. 숌은 이 시리즈를 항상 차례대로 만들고, 다른 영화는 만들지 않는다.

- 입력

첫째 줄에 숫자 N이 주어진다. N은 10,000보다 작거나 같은 자연수이다.

- 출력

첫째 줄에 N번째 영화의 제목에 들어간 수를 출력한다.
```
예제 입력 1 
2
예제 출력 1 
1666
```

In [166]:
movie=[]
i=666
while len(movie)!=10000:
    if '666' in str(i):
        movie.append(i)
    i+=1

n=int(input())
print(movie[n-1])

 2


1666


## 2750
- 문제

N개의 수가 주어졌을 때, 이를 오름차순으로 정렬하는 프로그램을 작성하시오.

- 입력

첫째 줄에 수의 개수 N(1 ≤ N ≤ 1,000)이 주어진다. 둘째 줄부터 N개의 줄에는 숫자가 주어진다. 이 수는 절댓값이 1,000보다 작거나 같은 정수이다. 수는 중복되지 않는다.

- 출력

첫째 줄부터 N개의 줄에 오름차순으로 정렬한 결과를 한 줄에 하나씩 출력한다.
```
예제 입력 1 
5
5
2
3
4
1
예제 출력 1 
1
2
3
4
5
```

In [168]:
n=int(input())
num=[]
for _ in range(n):
    num.append(int(input()))

num.sort()

for n in num:
    print(n)

## 2751
- 문제

N개의 수가 주어졌을 때, 이를 오름차순으로 정렬하는 프로그램을 작성하시오.

- 입력

첫째 줄에 수의 개수 N(1 ≤ N ≤ 1,000,000)이 주어진다. 둘째 줄부터 N개의 줄에는 숫자가 주어진다. 이 수는 절댓값이 1,000,000보다 작거나 같은 정수이다. 수는 중복되지 않는다.

- 출력

첫째 줄부터 N개의 줄에 오름차순으로 정렬한 결과를 한 줄에 하나씩 출력한다.
```
예제 입력 1 
5
5
4
3
2
1
예제 출력 1 
1
2
3
4
5
```

In [174]:
import sys
n=int(input())
num=[]
for _ in range(n):
    num.append(int(sys.stdin.readline()))

num.sort()

for n in num:
    print(n)

 5
 5
 4
 3
 2
 1


1
2
3
4
5


## 10989
- 문제

N개의 수가 주어졌을 때, 이를 오름차순으로 정렬하는 프로그램을 작성하시오.

- 입력

첫째 줄에 수의 개수 N(1 ≤ N ≤ 10,000,000)이 주어진다. 둘째 줄부터 N개의 줄에는 숫자가 주어진다. 이 수는 10,000보다 작거나 같은 자연수이다.

- 출력

첫째 줄부터 N개의 줄에 오름차순으로 정렬한 결과를 한 줄에 하나씩 출력한다.
```
예제 입력 1 
10
5
2
3
1
4
2
3
5
1
7
예제 출력 1 
1
1
2
2
3
3
4
5
5
7
```

In [ ]:
# 모름. 나중에

## 2108
- 문제

수를 처리하는 것은 통계학에서 상당히 중요한 일이다. 통계학에서 N개의 수를 대표하는 기본 통계값에는 다음과 같은 것들이 있다. 단, N은 홀수라고 가정하자.

산술평균 : N개의 수들의 합을 N으로 나눈 값
중앙값 : N개의 수들을 증가하는 순서로 나열했을 경우 그 중앙에 위치하는 값
최빈값 : N개의 수들 중 가장 많이 나타나는 값
범위 : N개의 수들 중 최댓값과 최솟값의 차이
N개의 수가 주어졌을 때, 네 가지 기본 통계값을 구하는 프로그램을 작성하시오.
- 입력

첫째 줄에 수의 개수 N(1 ≤ N ≤ 500,000)이 주어진다. 그 다음 N개의 줄에는 정수들이 주어진다. 입력되는 정수의 절댓값은 4,000을 넘지 않는다.
- 출력

첫째 줄에는 산술평균을 출력한다. 소수점 이하 첫째 자리에서 반올림한 값을 출력한다.
둘째 줄에는 중앙값을 출력한다.
셋째 줄에는 최빈값을 출력한다. 여러 개 있을 때에는 최빈값 중 두 번째로 작은 값을 출력한다.
넷째 줄에는 범위를 출력한다.
```
예제 입력 1 
5
1
3
8
-2
2
예제 출력 1 
2
2
1
10
예제 입력 2 
1
4000
예제 출력 2 
4000
4000
4000
0
예제 입력 3 
5
-1
-2
-3
-1
-2
예제 출력 3 
-2
-2
-1
2
```

In [182]:
# import sys
# n=int(sys.stdin.readline())
# a=[]
# for _ in range(n):
#     a.append(int(sys.stdin.readline()))
    

n=int(input())
a=[]
for _ in range(n):
    a.append(int(input()))

# a=[1,3,8,-2,2]

if sum(a)/len(a) >=0:
    if (sum(a)/len(a))%1 >= 0.5:       #평균 반올림
        print(int((sum(a)/len(a)))+1)
    else :
        print(int((sum(a)/len(a))))
else :
    if (sum(a)/len(a))%-1 >= -0.5:
        print(int((sum(a)/len(a))))
    else :
        print(int((sum(a)/len(a)))-1)

a.sort()
print(a[int(len(a)/2)]) # 2.중앙값

count=[]        #최빈값
for n in a:
    count.append(a.count(n))

max_count=max(count)

mode=[]
if count.count(max_count) > 2:
    for i in range(len(a)):
        if count[i]==max_count:
            mode.append(a[i])
    mode=list(set(mode))
    mode.sort()
    if len(mode)==1:
        print(mode[0])
    else :
        print(mode[1])
else : 
    print(a[count.index(max_count)])
    
print(max(a)-min(a))  # 범위

 5
 -1
 -2
 -3
 -1
 -2


-2
-2
-1
2


## 1427
- 문제

배열을 정렬하는 것은 쉽다. 수가 주어지면, 그 수의 각 자리수를 내림차순으로 정렬해보자.

- 입력

첫째 줄에 정렬하고자하는 수 N이 주어진다. N은 1,000,000,000보다 작거나 같은 자연수이다.

- 출력

첫째 줄에 자리수를 내림차순으로 정렬한 수를 출력한다.
```
예제 입력 1 
2143
예제 출력 1 
4321
```

In [20]:
a=input()

alist=[]
for n in a:
    alist.append(int(n))

alist.sort(reverse=True)

stra=''
for n in alist:
    stra += str(n)

print(stra)

 2143


4321


## 11650, 11651
- 문제

2차원 평면 위의 점 N개가 주어진다. 좌표를 x좌표가 증가하는 순으로, x좌표가 같으면 y좌표가 증가하는 순서로 정렬한 다음 출력하는 프로그램을 작성하시오.

- 입력

첫째 줄에 점의 개수 N (1 ≤ N ≤ 100,000)이 주어진다. 둘째 줄부터 N개의 줄에는 i번점의 위치 xi와 yi가 주어진다. (-100,000 ≤ xi, yi ≤ 100,000) 좌표는 항상 정수이고, 위치가 같은 두 점은 없다.

- 출력

첫째 줄부터 N개의 줄에 점을 정렬한 결과를 출력한다.
```
예제 입력 1 
5
3 4
1 1
1 -1
2 2
3 3
예제 출력 1 
1 -1
1 1
2 2
3 3
3 4
```

In [118]:
import operator
import sys

n = int(input())
a=[]

for _ in range(n):
    b = list(map(int, input().split()))
    a.append(b)

a.sort(key=operator.itemgetter(1,0))

for x, y in a:
    print(x,y)

 5
 3 4
 1 1
 1 -1
 2 2
 3 3


1 -1
1 1
2 2
3 3
3 4


## 1181
- 문제

알파벳 소문자로 이루어진 N개의 단어가 들어오면 아래와 같은 조건에 따라 정렬하는 프로그램을 작성하시오.

길이가 짧은 것부터
길이가 같으면 사전 순으로
- 입력

첫째 줄에 단어의 개수 N이 주어진다. (1≤N≤20,000) 둘째 줄부터 N개의 줄에 걸쳐 알파벳 소문자로 이루어진 단어가 한 줄에 하나씩 주어진다. 주어지는 문자열의 길이는 50을 넘지 않는다.

- 출력

조건에 따라 정렬하여 단어들을 출력한다. 단, 같은 단어가 여러 번 입력된 경우에는 한 번씩만 출력한다.
```
예제 입력 1 
13
but
i
wont
hesitate
no
more
no
more
it
cannot
wait
im
yours
예제 출력 1 
i
im
it
no
but
more
wait
wont
yours
cannot
hesitate
```

In [206]:
n=int(input())
words=[]
for _ in range(n):
    words.append(input())

for c in sorted(set(words),key=lambda x:(len(x),x)):
    print(c)

 5
 but
 i
 wont
 hesitate
 no


i
no
but
wont
hesitate


## 10814
- 문제

온라인 저지에 가입한 사람들의 나이와 이름이 가입한 순서대로 주어진다. 이때, 회원들을 나이가 증가하는 순으로, 나이가 같으면 먼저 가입한 사람이 앞에 오는 순서로 정렬하는 프로그램을 작성하시오.

- 입력

첫째 줄에 온라인 저지 회원의 수 N이 주어진다. (1 ≤ N ≤ 100,000)

둘째 줄부터 N개의 줄에는 각 회원의 나이와 이름이 공백으로 구분되어 주어진다. 나이는 1보다 크거나 같으며, 200보다 작거나 같은 정수이고, 이름은 알파벳 대소문자로 이루어져 있고, 길이가 100보다 작거나 같은 문자열이다. 입력은 가입한 순서로 주어진다.

- 출력

첫째 줄부터 총 N개의 줄에 걸쳐 온라인 저지 회원을 나이 순, 나이가 같으면 가입한 순으로 한 줄에 한 명씩 나이와 이름을 공백으로 구분해 출력한다.
```
예제 입력 1 
3
21 Junkyu
21 Dohyun
20 Sunyoung
예제 출력 1 
20 Sunyoung
21 Junkyu
21 Dohyun
```

In [211]:
import operator

n=int(input())

a=[]
for i in range(n):
    x,y=input().split()
    temp=[i, int(x),y]
    a.append(temp)

a.sort(key=operator.itemgetter(1,0))

for aa in a:
    print(aa[1],aa[2])

## 15649
- 문제

자연수 N과 M이 주어졌을 때, 아래 조건을 만족하는 길이가 M인 수열을 모두 구하는 프로그램을 작성하시오.

1부터 N까지 자연수 중에서 중복 없이 M개를 고른 수열
- 입력

첫째 줄에 자연수 N과 M이 주어진다. (1 ≤ M ≤ N ≤ 8)

- 출력

한 줄에 하나씩 문제의 조건을 만족하는 수열을 출력한다. 중복되는 수열을 여러 번 출력하면 안되며, 각 수열은 공백으로 구분해서 출력해야 한다.

수열은 사전 순으로 증가하는 순서로 출력해야 한다.

In [17]:
n,m = map(int,input().split())
temp=[]

def seq(temp):
    if len(temp)==m:
        print(*temp)
        return
    
    for i in range(1,n+1):
        if i in temp:
            continue
        temp.append(i) 
        seq(temp)
        temp.pop()
        
seq(temp)

 3 3


1 2 3
1 3 2
2 1 3
2 3 1
3 1 2
3 2 1


## 15650

- 문제

자연수 N과 M이 주어졌을 때, 아래 조건을 만족하는 길이가 M인 수열을 모두 구하는 프로그램을 작성하시오.

>1부터 N까지 자연수 중에서 중복 없이 M개를 고른 수열

>고른 수열은 오름차순이어야 한다.

- 입력

첫째 줄에 자연수 N과 M이 주어진다. (1 ≤ M ≤ N ≤ 8)

- 출력

한 줄에 하나씩 문제의 조건을 만족하는 수열을 출력한다. 중복되는 수열을 여러 번 출력하면 안되며, 각 수열은 공백으로 구분해서 출력해야 한다.

수열은 사전 순으로 증가하는 순서로 출력해야 한다.

In [16]:
# n,m = map(int,input().split())
n=3
m=3
temp=[]

def seq(temp):
    if len(temp)==m:
        print(*temp)
        return

    for i in range(1,n+1):
        if i in temp:
            continue
        if len(temp)>0:
            if temp[-1] > i:
                continue
        temp.append(i)
        seq(temp)
        temp.pop()
        
seq(temp)

1 2 3


### 15651
- 문제

자연수 N과 M이 주어졌을 때, 아래 조건을 만족하는 길이가 M인 수열을 모두 구하는 프로그램을 작성하시오.

1부터 N까지 자연수 중에서 M개를 고른 수열
같은 수를 여러 번 골라도 된다.
- 입력

첫째 줄에 자연수 N과 M이 주어진다. (1 ≤ M ≤ N ≤ 7)

- 출력

한 줄에 하나씩 문제의 조건을 만족하는 수열을 출력한다. 중복되는 수열을 여러 번 출력하면 안되며, 각 수열은 공백으로 구분해서 출력해야 한다.

수열은 사전 순으로 증가하는 순서로 출력해야 한다.

In [35]:
n,m = map(int,input().split())
temp=[]

def seq(temp):
    if len(temp)==m:
        print(*temp)
        return
    
    for i in range(1,n+1):
        temp.append(i) 
        seq(temp)
        temp.pop()
        
seq(temp)

 3 2


1 1
1 2
1 3
2 1
2 2
2 3
3 1
3 2
3 3


## 15652
- 문제

자연수 N과 M이 주어졌을 때, 아래 조건을 만족하는 길이가 M인 수열을 모두 구하는 프로그램을 작성하시오.

1부터 N까지 자연수 중에서 M개를 고른 수열
같은 수를 여러 번 골라도 된다.
고른 수열은 비내림차순이어야 한다.
길이가 K인 수열 A가 A1 ≤ A2 ≤ ... ≤ AK-1 ≤ AK를 만족하면, 비내림차순이라고 한다.
- 입력

첫째 줄에 자연수 N과 M이 주어진다. (1 ≤ M ≤ N ≤ 8)

- 출력

한 줄에 하나씩 문제의 조건을 만족하는 수열을 출력한다. 중복되는 수열을 여러 번 출력하면 안되며, 각 수열은 공백으로 구분해서 출력해야 한다.

수열은 사전 순으로 증가하는 순서로 출력해야 한다.

In [39]:
n,m = map(int,input().split())
temp=[]

def seq(temp):
    if len(temp)==m:
        print(*temp)
        return
    
    for i in range(1,n+1):
        if len(temp)>0:
            if temp[-1] > i:
                continue
        temp.append(i) 
        seq(temp)
        temp.pop()
        
seq(temp)

 3 2


1 1
1 2
1 3
2 2
2 3
3 3


## 9663
- 문제

N-Queen 문제는 크기가 N × N인 체스판 위에 퀸 N개를 서로 공격할 수 없게 놓는 문제이다.

N이 주어졌을 때, 퀸을 놓는 방법의 수를 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 N이 주어진다. (1 ≤ N < 15)

- 출력

첫째 줄에 퀸 N개를 서로 공격할 수 없게 놓는 경우의 수를 출력한다.
```
예제 입력 1 
8
예제 출력 1 
92
```

In [98]:
n=int(input())

row = []
cnt = 0
def tf(row, col):
    for i, r in enumerate(row):
        if r==col or abs(len(row)-i)==abs(r-col):
            return True
    return False

def seq(row):
    if len(row)==n:
        print(row)
        global cnt
        cnt += 1
        return
    
    for col in range(n):
        if tf(row, col):
            continue
            
        row.append(col)
        seq(row)
        row.pop()
        
seq(row)
print(cnt)

 6


[1, 3, 5, 0, 2, 4]
[2, 5, 1, 4, 0, 3]
[3, 0, 4, 1, 5, 2]
[4, 2, 0, 5, 3, 1]
4


## 14888
- 문제

N개의 수로 이루어진 수열 A1, A2, ..., AN이 주어진다. 또, 수와 수 사이에 끼워넣을 수 있는 N-1개의 연산자가 주어진다. 연산자는 덧셈(+), 뺄셈(-), 곱셈(×), 나눗셈(÷)으로만 이루어져 있다.

우리는 수와 수 사이에 연산자를 하나씩 넣어서, 수식을 하나 만들 수 있다. 이때, 주어진 수의 순서를 바꾸면 안 된다.

예를 들어, 6개의 수로 이루어진 수열이 1, 2, 3, 4, 5, 6이고, 주어진 연산자가 덧셈(+) 2개, 뺄셈(-) 1개, 곱셈(×) 1개, 나눗셈(÷) 1개인 경우에는 총 60가지의 식을 만들 수 있다. 예를 들어, 아래와 같은 식을 만들 수 있다.
```
1+2+3-4×5÷6
1÷2+3+4-5×6
1+2÷3×4-5+6
1÷2×3-4+5+6
```
식의 계산은 연산자 우선 순위를 무시하고 앞에서부터 진행해야 한다. 또, 나눗셈은 정수 나눗셈으로 몫만 취한다. 음수를 양수로 나눌 때는 C++14의 기준을 따른다. 즉, 양수로 바꾼 뒤 몫을 취하고, 그 몫을 음수로 바꾼 것과 같다. 이에 따라서, 위의 식 4개의 결과를 계산해보면 아래와 같다.
```
1+2+3-4×5÷6 = 1
1÷2+3+4-5×6 = 12
1+2÷3×4-5+6 = 5
1÷2×3-4+5+6 = 7
```
N개의 수와 N-1개의 연산자가 주어졌을 때, 만들 수 있는 식의 결과가 최대인 것과 최소인 것을 구하는 프로그램을 작성하시오.

- 입력

첫째 줄에 수의 개수 N(2 ≤ N ≤ 11)가 주어진다. 둘째 줄에는 A1, A2, ..., AN이 주어진다. (1 ≤ Ai ≤ 100) 셋째 줄에는 합이 N-1인 4개의 정수가 주어지는데, 차례대로 덧셈(+)의 개수, 뺄셈(-)의 개수, 곱셈(×)의 개수, 나눗셈(÷)의 개수이다. 

- 출력

첫째 줄에 만들 수 있는 식의 결과의 최댓값을, 둘째 줄에는 최솟값을 출력한다. 최댓값과 최솟값이 항상 -10억보다 크거나 같고, 10억보다 작거나 같은 결과가 나오는 입력만 주어진다. 또한, 앞에서부터 계산했을 때, 중간에 계산되는 식의 결과도 항상 -10억보다 크거나 같고, 10억보다 작거나 같다.
```
예제 입력 1 
2
5 6
0 0 1 0
예제 출력 1 
30
30
```

In [235]:
a=int(input())
nums=list(map(int, input().split()))
ari=list(map(int, input().split()))
temp=[]
ans=[]
def arithmetic(nums, temp):
    res=nums[0]
    for i,op in enumerate(temp):
        if op==0:
            res += nums[i+1]
        if op==1:
            res -= nums[i+1]
        if op==2:
            res *= nums[i+1]
        if op==3:
            if res<0 and nums[i+1] >0:
                res = (abs(res)//nums[i+1])*-1            
            else:
                res //= nums[i+1]
    return res


def seq():
    if len(temp)==sum(ari):
        res=arithmetic(nums, temp)
        ans.append(res)
        return 
        
    for i in range(len(ari)):
        if temp.count(i) < ari[i]:
            temp.append(i)
            seq()
            temp.pop()      
seq()
print(max(ans))
print(min(ans))

 3
 3 4 5
 1 0 1 0


35
17
